File: go/LeetCode/coinChange.go
package main

func main() {
	
}

func coinChange(coins []int, amount int) int {
	max:=amount+1
	dp := make([]int, amount+1)
	for i,_ :=range dp{
		dp[i] = max
	}
	dp[0]=0
	for _,v := range coins {
		for j:=v;j<=amount;j++ {
			dp[j] = min(dp[j], dp[j-v]+1)
		}
	}
	if dp[amount]==max {
		return -1
	}else {
		return dp[amount]
	}
}

File: cdd/LeetCode/LeetCode/coinChange.cpp
#include <vector>
#include <stdio.h>

using namespace std;

class solution {
public:
	int coinChange(vector<int>& coins, int amount) {
		vector<int> dp = vector<int>(amount + 1, amount+1);
		dp[0] = 0;
		for (int i = 0; i < coins.size(); i++)
		{
			for (int j = coins[i]; j <= amount; j++) 
			{
				dp[j] = min(dp[j], dp[j - coins[i]] + 1);
			}
		}
		if (dp[amount] == amount+1)
		{
			return -1;
		}
		else {
			return dp[amount];
		}
	
	}

};